core/templates/dev/head/pages/community-dashboard-page/services/translate-text-backend-api.service.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service to make http calls present inside servicees related
+ * to community-dashboard-page services
+ */
+
+angular.module('oppia').factory('TranslateTextBackendService', [
+  '$http', '$q',
+  function(
+      $http, $q) {
+    var fetchTranslatableTextHandler = function(params, successCallback,
```

Comment:
Hmm, I'm thinking we should drop the "Handler" suffix here and elsewhere, since it makes it sound like one is retrieving a handler function

Reply:
Agreed. I named it according to the endpoint it hits. I'll change the name. Any suggestions on what I should name it?

Reply:
I suggest taking a look at how other services do it.

Also, please don't use "params" here. Separate the args out into meaningful separate arguments (pass each arg separately into this function) and use this method to combine them into the correct payload.

Reply:
Sure, will do that.

Reply:
@seanlip what should I do when there are simply too many params?

Reply:
Still break them down into individual args and assemble them here.

Reply:
Won't that affect readability badly?

Reply:
No.

Reply:
I removed the handler suffix and am now passing separate variables instead of a params object